### PR TITLE
perf(http): reduce data transfer when using HTTP caching

### PR DESCRIPTION
**Pull request by alan-agius4**
This commit reduces the property size in the http transfer cache to reduce the page payload.

Before
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "body": "....",
    "headers": {},
    "status": 200,
    "statusText": "OK",
    "url": "http://foo.com/assets/media.json",
    "responseType": "json"
  },
}
</script>
```

Now
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "b": "....",
    "h": {},
    "s": 200,
    "st": "OK",
    "u": "http://foo.com/assets/media.json",
    "rt": "json"
  },
}
</script>
```

### CODE REVIEW

**Comment by HyperLife1119** [comment]
How about using tuples?

**Comment by alan-agius4** [comment]
> How about using tuples?

I went with a similar approach that is used for other things like hydration. Also with tuples, it feel like it would be harder to visually read what is written in the cache.

File: packages/common/http/src/transfer_cache.ts [hunk 1]
```diff
@@ -36,12 +36,18 @@ export type HttpTransferCacheOptions = {
...
 
 interface TransferHttpResponse {
-  body: any;
-  headers: Record<string, string[]>;
-  status?: number;
-  statusText?: string;
-  url?: string;
-  responseType?: HttpRequest<unknown>['responseType'];
+  /** body */
+  b: any;
```
Comment (AndrewKushnir) [inline packages/common/http/src/transfer_cache.ts:40]: I think it'd be great to have those keys in consts, something similar to this:

https://github.com/angular/angular/blob/main/packages/core/src/hydration/interfaces.ts#L27-L38

**Review by AndrewKushnir — APPROVED** [review]
Looks great! 👍

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit c5e30f1d79f3ebd85cb8061452a897313208bbca.

**Pull request closed without merging**